src/kvstore/plugins/hbase/HBaseStore.h

```diff
@@ -0,0 +1,114 @@
+#define KVSTORE_HBASESTORE_H_
+
+#include "base/Base.h"
+#include "kvstore/KVStore.h"
+#include "kvstore/PartManager.h"
+#include "kvstore/KVEngine.h"
+#include <gtest/gtest_prod.h>
+
+namespace nebula {
+namespace kvstore {
+
+class HBaseStore : public KVStore {
+    FRIEND_TEST(HBaseStoreTest, SimpleTest);
+
+public:
+    explicit HBaseStore(KVOptions options)
+            : options_(std::move(options)) {
+        schemaMan_ = std::move(options_.schemaMan_);
+        partMan_ = std::move(options_.partMan_);
+    }
+
+    ~HBaseStore() = default;
+
+    /**
+     * Pull meta information from PartManager and init current instance.
+     * */
+    void init();
+
+    uint32_t capability() const override {
+        return 0;
+    }
+
+    ResultCode get(GraphSpaceID spaceId,
+                   PartitionID  partId,
+                   const std::string& key,
+                   std::string* value) override;
+
+    ResultCode multiGet(GraphSpaceID spaceId,
+                        PartitionID partId,
+                        const std::vector<std::string>& keys,
+                        std::vector<std::string>* values) override;
+
+    /**
+     * Get all results in range [start, end)
+     * */
+    ResultCode range(GraphSpaceID spaceId,
+                     PartitionID  partId,
+                     const std::string& start,
+                     const std::string& end,
+                     std::unique_ptr<KVIterator>* iter) override;
+
+    /**
+     * Get all results with prefix.
+     * */
+    ResultCode prefix(GraphSpaceID spaceId,
+                      PartitionID  partId,
+                      const std::string& prefix,
+                      std::unique_ptr<KVIterator>* iter) override;
+
+    /**
+     * async batch put.
+     * */
+    void asyncMultiPut(GraphSpaceID spaceId,
+                       PartitionID  partId,
+                       std::vector<KV> keyValues,
+                       KVCallback cb) override;
+
+    void asyncRemove(GraphSpaceID spaceId,
+                     PartitionID partId,
+                     const std::string& key,
+                     KVCallback cb) override;
+
+    void asyncMultiRemove(GraphSpaceID spaceId,
+                          PartitionID  partId,
+                          std::vector<std::string> keys,
+                          KVCallback cb) override;
+
+    void asyncRemoveRange(GraphSpaceID spaceId,
+                          PartitionID partId,
+                          const std::string& start,
+                          const std::string& end,
+                          KVCallback cb) override;
+
+
+private:
+    void addSpace(GraphSpaceID spaceId);
+
+    void removeSpace(GraphSpaceID spaceId);
+
+    Engine newEngine(GraphSpaceID spaceId);
+
+private:
+    std::unordered_map<GraphSpaceID, Engine> kvs_;
+
+    folly::RWSpinLock lock_;
+
+    std::unique_ptr<meta::SchemaManager> schemaMan_{nullptr};
+
+    std::unique_ptr<PartManager> partMan_{nullptr};
```

Comment:
ditto

Reply:
The **schemaMan_** is used to obtain the scheam of Tag and EdgeType. When data is obtained or stored from HBase, scheam is needed to encode and decode the data.

The **partMan_** is used to obtain the partsMap, which can be used to know which GraphSpace is included in the Nebula cluster, and then initialize the GraphSpace to create the corresponding HBaseEngine.

Reply:
Ditto.  I don't think we really need partMan_

Reply:
done!